fv3core/stencils/remapping_part2.py

```diff
@@ -27,125 +27,176 @@ def init_phis(hs: FloatField, delz: FloatField, phis: FloatField, te_2d: FloatFi
-                + str(consv)
-                + ")  < -CONSV_MIN("
-                + str(-constants.CONSV_MIN)
-                + ")"
-            )
+class Remapping_Part2:
+    def __init__(self):
+        self.grid = spec.grid
+        self.namelist = spec.namelist
+        self._saturation_adjustment = utils.cached_stencil_class(SatAdjust3d)(
+            cache_key="satadjust3d"
+        )
+        self._copy_from_below_stencil = FrozenStencil(
+            copy_from_below,
+            origin=self.grid.compute_origin(),
+            domain=self.grid.domain_shape_compute(),
+        )
 
-    if spec.namelist.do_sat_adj:
-
-        kmp_origin = (grid.is_, grid.js, kmp)
-        kmp_domain = (grid.nic, grid.njc, grid.npz - kmp)
-        saturation_adjustment(
-            te,
-            qvapor,
-            qliquid,
-            qice,
-            qrain,
-            qsnow,
-            qgraupel,
-            qcld,
-            hs,
-            peln,
-            delp,
-            delz,
-            q_con,
-            pt,
-            pkz,
-            cappa,
-            r_vir,
-            mdt,
-            fast_mp_consv,
-            last_step,
-            akap,
-            kmp,
+        self._moist_cv_last_step_stencil = FrozenStencil(
+            moist_cv.moist_pt_last_step,
+            origin=(self.grid.is_, self.grid.js, 0),
+            domain=(self.grid.nic, self.grid.njc, self.grid.npz + 1),
         )
-        if fast_mp_consv:
-            sum_te(te, te0_2d, origin=kmp_origin, domain=kmp_domain)
-    if last_step:
-        moist_cv.compute_last_step(
-            pt, pkz, dtmp, r_vir, qvapor, qliquid, qice, qrain, qsnow, qgraupel, gz
+
+        self._basic_adjust_divide_stencil = FrozenStencil(
+            basic.adjust_divide_stencil,
+            origin=self.grid.compute_origin(),
+            domain=self.grid.domain_shape_compute(),
         )
-    else:
-        basic.adjust_divide_stencil(
-            pkz, pt, origin=grid.compute_origin(), domain=grid.domain_shape_compute()
+
+        self.do_sat_adjust = self.namelist.do_sat_adj
+
+        # TODO : When remapping becomes an object, these stencils can be declared in
+        #        __init__ when (I think) pFull can be passed into __init__
+        # self._saturation_adjustment_stencil = FrozenStencil(SatAdjust3d,...)
+        # self._sum_te_stencil = FrozenStencil(sum_te,...)
+
+        # TODO : When state.pt from fv_dynamics can be passed into __init__, these
+        #        storages can be declared in __init__
+        # phis = utils.make_storage_from_shape(
+        #     pt.shape, grid.compute_origin(), cache_key="remapping_part2_phis"
+        # )
+        # te_2d = utils.make_storage_from_shape(
+        #     pt.shape[0:2], grid.compute_origin(), cache_key="remapping_part2_te_2d"
+        # )
+        # zsum1 = utils.make_storage_from_shape(
+        #     pt.shape[0:2], grid.compute_origin(), cache_key="remapping_part2_zsum1"
+        # )
```

Comment:
since the shapes of all the fields are the same, you can get the shape from grid.domain_shape_compute(add=(1,1,1)). using the other fields shape was just done for convenience. SO you can declare the temporaries here, yay!

Reply:
It occurred to me that these storages `phi`, `te_2d`, and `zsum1` are not used anywhere.  I deleted them in `__call__`, and the local tests still pass...